### PR TITLE
BOAC-63, cohortFactory.getCohorts() is called when user is not authenticated

### PR DESCRIPTION
**Pull request by johncrossman**
https://jira.ets.berkeley.edu/jira/browse/BOAC-63

See Jira description.

### CODE REVIEW

File: boac/static/app/auth/authFactory.js [hunk 1]
```diff
@@ -26,11 +24,12 @@
...
         function successCallback() {
-          $rootScope.$emit('authenticationSuccess');
-          refreshStatus();
+          refreshStatus().then(function() {
+            $rootScope.$broadcast('userStatusChange');
+          });
         },
         function errorCallback() {
-          $rootScope.$emit('authenticationFailure');
+          $rootScope.$broadcast('authenticationFailure');
```
Comment (johncrossman) [inline boac/static/app/auth/authFactory.js]: The confusing art of event listening: https://toddmotto.com/all-about-angulars-emit-broadcast-on-publish-subscribing/

File: boac/static/app/bootstrap.js [hunk 2]
```diff
@@ -6,7 +6,7 @@
     var initInjector = angular.injector([ 'ng' ]);
     var $http = initInjector.get('$http');
 
-    return $http.get('/api/status').then(function(results) {
+    return $http.get('/api/config').then(function(results) {
```
Comment (johncrossman) [inline boac/static/app/bootstrap.js]: Unrelated bug. How did this happen?!

File: boac/static/app/routes.js [hunk 3]
```diff
@@ -31,15 +31,6 @@
...
         controller: 'StudentController'
       });
-
-  }).run(function($rootScope) {
-    var initInjector = angular.injector([ 'ng' ]);
-    var $http = initInjector.get('$http');
-
-    return $http.get('/api/status').then(function(results) {
-      $rootScope.me = results.data;
-    });
```
Comment (johncrossman) [inline boac/static/app/routes.js]: broadcasting events from the depths of angular `module()` was not working. IMO, having this init in `authService.js` is equally intuitive and easier to find.

**Comment by paulkerschen** [comment]
Thanks for the education on `$emit` vs. `$broadcast`, @johncrossman. Moving auth into a service makes sense to me.

**Pull request merged**